src/services/doppler-api-client.test.ts

```diff
@@ -528,7 +530,102 @@ describe('HttpDopplerApiClient', () => {
+      const dopplerApiClient = createHttpDopplerApiClient({ request });
+
+      // Act
+      const result = await dopplerApiClient.getSubscriberFieldHistory({
+        subscriberEmail: 'a@a.com',
+        fieldName: 'MiPermiso',
+      });
+
+      // Assert
+      expect(request).toBeCalledTimes(1);
+      expect(result).not.toBe(undefined);
+      expect(result.success).toBe(false);
+    });
+
+    it('should call with the right parameters and parse successful response', async () => {
+      // Arrange
+      const subscriberEmail = 'a@a.com';
+      const fieldName = 'MiPermiso';
+      const fieldsResult = {
+        data: {
+          items: [
+            {
+              subscriberEmail: 'email@a.com',
+              fieldName: 'fieldName',
+              fieldType: 'permission',
+              date: '2021-02-10T15:22:00.000Z',
+              value: 'true',
+              originIP: '181.167.226.47',
+              originType: 'Formulario',
+            },
+            {
+              subscriberEmail: 'email@a.com',
+              fieldName: 'fieldName',
+              fieldType: 'permission',
+              date: '2021-02-05T10:11:24.000Z',
+              value: 'true',
+              originIP: '181.167.226.30',
+              originType: 'Formulario',
+            },
+            {
+              subscriberEmail: 'email@a.com',
+              fieldName: 'fieldName',
+              fieldType: 'permission',
+              date: '2021-01-05T01:05:04.123Z',
+              value: 'true',
+              originIP: '181.167.226.20',
+              originType: 'Manual',
+            },
+          ],
+          currentPage: 1,
+          itemsCount: 3,
+          pagesCount: 1,
+          _links: [],
+        },
+      };
+      const request = jest.fn(async () => fieldsResult);
+      const dopplerApiClient = createHttpDopplerApiClient({ request });
+
+      // Act
+      const result = await dopplerApiClient.getSubscriberFieldHistory({
+        subscriberEmail,
+        fieldName,
+      });
+
+      // Assert
+      expect(request).toBeCalledTimes(1);
+      expect(request).toBeCalledWith({
+        headers: {
+          Authorization: `token ${jwtToken}`,
+        },
+        method: 'GET',
+        url: `/accounts/${accountEmail}/subscribers/${subscriberEmail}/fields/${fieldName}/history`,
+      });
+      expect(result).not.toBe(undefined);
+      expect(result.success).toBe(true);
+      expect(result.success && result.value).not.toBe(null);
+      expect(result.success && result.value.items).toHaveLength(3);
+      for (var item of (result.success && result.value.items) || []) {
+        expect(item).toMatchObject({
+          date: expect.any(Date),
+          fieldName: expect.any(String),
+          fieldType: expect.any(String),
+          originIP: expect.any(String),
+          originType: expect.any(String),
+          subscriberEmail: expect.any(String),
+          value: expect.any(String),
+        });
+      }
```

Comment:
Do you know a better way to do this with Jest? I mean, I would prefer to remove the foreach and using something like this:

```suggestion
      expect((result.success && result.value.items) || []).All.toMatchObject({
        date: expect.any(Date),
        fieldName: expect.any(String),
        fieldType: expect.any(String),
        originIP: expect.any(String),
        originType: expect.any(String),
        subscriberEmail: expect.any(String),
        value: expect.any(String),
      });
```

Reply:
I don´t know sorry...